app/front/scripts/controllers/main.js

```diff
@@ -69,7 +73,24 @@ angular.module('Application')
       function initRegular() {
         $q(LoginService.tryGetToken())
           .then(function(token) {
-            return $q(osViewerService.loadDataPackages(token));
+            var packageId = null;
+            var userId = token ? LoginService.getUserId() : null;
+            if (!token) {
```

Comment:
Well, we should see this dropdown really as a 'my datasets' navigation control.
If the user is not logged in, this control simply doesn't appear. Either way, it should show only the user's datasets and not some other people's stuff.
I know that right now it's somewhat central in the program flow, but I think it would be best to make the necessary refactoring to take it out of the flow and really treat is as an independent component.